src/cobalt.rs

```diff
@@ -120,6 +124,57 @@ pub fn build(config: &Config) -> Result<()> {
         }
     }
 
+    // load data files
+    let mut data_map : HashMap<String, Value> = HashMap::new();
+    let data_root = source.join("_data");
+    let data_files_builder = FilesBuilder::new(data_root.as_path())?;
+    let data_files = data_files_builder.build()?;
+
+    for df in data_files.files() {
+
+        if let Some(ext) = df.extension() {
+            if let Some(file_stem) = df.file_stem() {
+                let file_name = String::from(file_stem.to_str().unwrap());
+                let full_path = data_root.join(df.clone());
+                let data : Value;
+
+                if ext == OsStr::new("yml") || ext == OsStr::new("yaml")  {
+                    let reader = File::open(full_path)?;
+                    data = serde_yaml::from_reader(reader)?;
+                } else if ext == OsStr::new("json") {
+                    let reader = File::open(full_path)?;
+                    data = serde_json::from_reader(reader).unwrap();
+                } else if ext == OsStr::new("toml") {
+                    let mut reader = File::open(full_path)?;
+                    let mut text = String::new();
+                    reader.read_to_string(&mut text)?;
+                    data = toml::from_str(&text).unwrap();
+                } else {
+                    continue
+                }
+
+                // now find the nested map it is supposed to be in
+                let mut target_map = if let Some(path) = df.parent() {
+                    let mut map = &mut data_map;
+                    for part in path.to_str().unwrap().split("/") {
+                        if part.len() == 0 { continue }
+                        let mut cur_map = map;
+                        map = cur_map
+                            .entry(String::from(part))
+                            .or_insert_with(|| Value::Object(HashMap::new()))
+                            .as_object_mut().unwrap();
+                    }
+                    map
+                } else {
+                    &mut data_map
+                };
+
+                target_map.insert(file_name, data);
```

Comment:
> merging data at the same point rather than overwrite it

I assume the only way we can overwrite is if 
- A file stem matches a directory name which contains more data
- Two files have the same stem but different extensions

Ignoring or overwriting will be surprising to people.  We should merge or error.  I lean towards erroring.

Reply:
That is exactly the problem, having a folder with `_data/titles/*yaml` and `_data/titles.yaml` or any other combination of supported files. Overwriting isn't my preference, was just easier to implement 😀. But I agree erroring is better. Will implement that.

Reply:
I was actually wrong. [`HashMap.insert` ignores the new value, if the key already exists](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.insert). But using that I made a version that prints an error message to the user that it was skipped. I opted for not hard-error on that...

Reply:
> If the map did have this key present, the value is updated, and the old value is returned. The key is not updated,

Sounds like it doesn't ignore the new value, only the new key is ignored.  For example, if your key type is a case insensitive string then you can have two keys with different binary representations that are equal.  If you insert a `("key", "A")` followed by `("kEy", "B")`, then the map should contain `("key", "B")`.

Reply:
Oh, true, I misread that. Guess I have to make an explicit check before adding it then... meeeh... not cool.